### ISSUE TITLE
Query optimalization issue

### ISSUE THREAD

**Issue by grdw** [quintel/etmoses#1309]
![screen shot 2016-07-20 at 14 15 33](https://cloud.githubusercontent.com/assets/2676542/16985997/80fb5a6e-4e84-11e6-9818-03f6f1bb3abd.png)

I think this is called: spamming the MySQL server. Why can't we query a LoadProfile once, keep it in memory and than request it again from memory? Instead of continiously asking the MySQL server. Also I think all these profiles can be queried in a single go.

**Comment by antw** [comment]
I agree that this can be improved, but the important word on each line is `CACHE`. It's not hitting MySQL each time, but instead loading the record from an [identity map](http://api.rubyonrails.org/v3.2.5/classes/ActiveRecord/IdentityMap.html).
